### PR TITLE
Add a /metrics endpoint

### PR DESCRIPTION
**Pull request by rozap**
* Added a metrics class which samples the memory
* Added a middleware bit to the main endpoints to
  track http response codes
* Remove the interval spam

Refs EN-3115

### CODE REVIEW

File: es6-lib/util/metrics.js [hunk 1]
```diff
@@ -9,12 +9,59 @@ var conf = config();
...
+        status: {}
+      },
+      gc: {},
+      memory: {}
+    };
+  }
+
+  metrics(req, res) {
+    res.status(200).send(JSON.stringify(this.state));
+    this._resetState();
```
Comment (michaelb990) [inline es6-lib/util/metrics.js]: Are the metrics contained here instantaneous samples or diffs from the last time metrics were collected? Basically, will the metrics returned be affected by the time elapsed between calls?

Comment (rozap) [inline es6-lib/util/metrics.js]: my understanding is that collectd handles that. they're reset each time `/metrics` is called. it looks like collectd will divide the count by the time between samples in order to get the rate; https://collectd.org/wiki/index.php/Data_source

Comment (michaelb990) [inline es6-lib/util/metrics.js]: Sure, but if you're using a counter for requests, you want it to keep going up, not be reset between each call to `/metrics`, right?

File: es6-lib/util/metrics.js [hunk 2]
```diff
@@ -9,12 +9,59 @@ var conf = config();
 
 class Metrics {
   constructor() {
-    setInterval(this._sampleMemory.bind(this), 1000);
+    this._resetState();
     this._gc = gcstats();
     this._gc.on('stats', _.throttle(this._onGcStat, 1000).bind(this));
```
Comment (michaelb990) [inline es6-lib/util/metrics.js]: what does this do?

Comment (rantler) [inline es6-lib/util/metrics.js]: Prevents thrashing by calling the `this._onGcStat ` function no more than once every 1,000ms.

**Pull request merged**